src/Products/GenericSetup/interfaces.py

```diff
@@ -535,9 +550,27 @@ def runAllImportStepsFromProfile(profile_id, purge_old=None,
         o Unless 'ignore_dependencies' is true this will also import
           all profiles this profile depends on.
 
+        o 'archive' is a tarball with files to import.
+          Note that metadata.xml is never read, so you cannot use dependencies.
+
         o 'blacklisted_steps' can be a list of step-names that won't be
           executed. Use with special care and only for special cases.
 
+        o 'dependency_strategy' is an integer describing what do we do with
+          already applied dependency profiles.  The default is to apply new
+          profiles (of course), and upgrade outdated ones.
+          See 'DEFAULT_DEPENDENCY_STRATEGY' in 'tool.py'
+
+        o 'path' is the path to a directory with files to import.
+          This avoids the need to register a full profile, when you only need
+          to import a few files once in an upgrade step.
+          It is best to use an absolute path, otherwise it would depend on
+          your working directory.
+          Note that metadata.xml is never read, so you cannot use dependencies.
+
```

Comment:
Add another place, you spoke about restricting *path* to a package path (for security reasons).

Reply:
That is in zcml. There we only accept a path within a package. This is then translated to an absolute directory. This is needed here, because you call `runAllImportStepsFromProfile(profile_id=None, path='...')` so the code here does not know which package the None profile belongs to.

So in Python code you could call `runAllImportStepsFromProfile(profile_id=None, path="/etc")` and try to do something clever with `/etc/passwd` in an import step.  But in any import step you can just call `open('/etc/passwd')` anyway, so this should not raise any further security concerns.